packages/syft/src/syft/service/network/network_service.py

```diff
@@ -211,7 +228,11 @@ def add_peer(
             )
 
         try:
-            remote_client: SyftClient = peer.client_with_context(context=context)
+            remote_client = peer.client_with_context(context=context)
+            if remote_client.is_err():
+                return SyftError(message=f"{remote_client.err()}")
```

Comment:
suggestion: Maybe we can update the error message to "Failed to create remote client for peer: {peer_id}. Error: {remote_client.err()}"